test/adjoints.jl

```diff
@@ -0,0 +1,28 @@
+using SymbolicUtils, SymbolicUtils.Code
+using Zygote
+
+@testset "create_array adjoint" begin
+  elems = (1,2,3,4,5,)
+
+  Ts = (Float64, Float32, Float16, Int64, Int32)
+  dims_candidates = ((1, (2,3)), (2, (1,3)))
+  As = (Array,)
+
+  for T in Ts,
+      dims in dims_candidates,
+      A in As
+
+    u, dim = dims
+    ŷ, pb = Zygote.pullback(elems) do elems
+      SymbolicUtils.Code.create_array(A, T, Val(u), Val(dim), elems...)
+    end
+    y = SymbolicUtils.Code.create_array(A, T, Val(u), Val(dim), elems...)
+    @test y == ŷ
+
+    gs = pb(ones(T, length(elems)))
+    @test length(gs[1]) == length(elems)
+    for i = 1:(prod(dim)-1)
+      @test gs[1][i] == one(eltype(ŷ))
+    end
+  end
+end 
```

Comment:
@AlCap23 and @doddgray would love to know if this is correct and works for your case.

Reply:
@shashi sorry for my delayed response. Thank you for checking in! So far Zygote gradients involving array-valued functions generated with `Symbolics.build_function(A::AbstractArray{Num},args...)` now seem to work and provide correct results, for my test cases. I still need to figure out how to properly use SymbolicUtils' code generation before I can take advantage. I see reasonable primal and Zygote.gradient evaluation performance when I generate a functions in the global scope, but functions which run `build_function` internally (using `expression=Val{false}` for `RuntimeGeneratedFunctions`) have very inefficient primals and gradients. I don't think this is a Zygote-specific issue because the primal evaluations suffer similarly to the gradients, rather I think the generated functions are being compiled over and over again?  Can you recommend a way to initialize/compile RGF's that will be evaluated many times used within a function? I ask here because I worry that Zygote performance might depend on the scope or module where the function is generated, but maybe not the right place...